### PR TITLE
Adding groovy discovery plugin and npm-groovy-lint tool plugins.

### PR DESCRIPTION
**Pull request by xydesa**

### CODE REVIEW

File: src/statick_web/plugins/tool/groovylint_tool_plugin.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                try:
+                    err_dict = json.loads(line)
+                    if "files" in err_dict:
+                        all_files = err_dict["files"]
+                        for file_name in all_files:
+                            file_errs = all_files[file_name]
+                            if "errors" in file_errs:
+                                for issue in file_errs["errors"]:
+                                    severity_str = issue["severity"]
+                                    severity = 3
```
Comment (denewiler) [inline src/statick_web/plugins/tool/groovylint_tool_plugin.py:93]:
```suggestion
                                    severity = "3"
```

For better or worse (I think worse) the severity is a string. Maybe someday we'll change it.

https://github.com/sscpac/statick/blob/main/statick_tool/issue.py#L15

File: tests/tool/groovylint_tool_plugin/test_groovylint_tool_plugin.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    issues = plugin.parse_output(output)
+    assert not issues
+
+
+@mock.patch("statick_tool.plugins.tool.groovylint_tool_plugin.subprocess.check_output")
+def test_groovylint_tool_plugin_scan_calledprocesserror(mock_subprocess_check_output):
+    """
+    Test what happens when a CalledProcessError is raised (usually means groovylint hit an error).
+
+    Expected result: issues is None
```
Comment (denewiler) [inline tests/tool/groovylint_tool_plugin/test_groovylint_tool_plugin.py:132]:
```suggestion
    Expected result: no issues found
```

File: src/statick_web/plugins/discovery/groovy_discovery_plugin.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        """Scan package looking for Groovy files."""
+        src_files: List[str] = []
+
+        self.find_files(package)
+
+        for file_dict in package.files.values():
+            if (
+                file_dict["name"].endswith(".groovy")
+                or file_dict["name"].endswith(".gradle")
+                or file_dict["name"].startswith("jenkinsfile")
```
[the remark this answers is not in the document]
Comment (xydesa) [reply src/statick_web/plugins/discovery/groovy_discovery_plugin.py:31]: The filename is already put in lowercase by the DiscoveryPlugin superclass: https://github.com/sscpac/statick/blob/main/statick_tool/discovery_plugin.py#L51

**Comment by xydesa** [comment]
Closing in favor of adding these plugins to the main statick repo: https://github.com/sscpac/statick/pull/370

**Pull request closed without merging**